src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -844,65 +919,75 @@ protected async Task AppendAsyncInternal(DateTime timestamp, ICollection<double>
             {
                 using (_operation.ConcurrencyCheck())
                 {
-                    await _operation.ExecuteBeforeStore().ConfigureAwait(false);
-
+                    var strLock = await _operation._streamLock.ReaderLockAsync().ConfigureAwait(false);
```

Comment:
this is already disposable, isn't it better to use `using` ?

Reply:
not relevant anymore